src/py/pyodide/http.py

```diff
@@ -40,6 +40,9 @@ def __init__(self, status: int, status_text: str, url: str) -> None:
             super().__init__(f"{status} Client Error: {status_text} for url: {url}")
         elif 500 <= status < 600:
             super().__init__(f"{status} Server Error: {status_text} for url: {url}")
+        raise ValueError(
+            f"Invalid error code not comprised between 400 and 599: {status}"
+        )
```

Comment:
I think it's a bad practice to raise an error from the constructor of an error if it can be avoided because it will overwrite the original error context. Instead, why not put a message that says that it's an unexpected status code?

Reply:
Sure. I dont' like being able to construct an invalid error, but that would works as well.